tools/ncf.py

```diff
@@ -489,3 +424,114 @@ def generate_technique_content(technique_metadata):
 
   return result
 
+
+# FUNCTIONS called directly by the API code
+###########################################
+
+def get_all_techniques_metadata(include_methods_calls = True, alt_path = ''):
+  all_metadata = {}
+
+  if alt_path != '': sys.stderr.write("INFO: Alternative source path added: %s\n" % alt_path)
+
+  filenames = get_all_techniques_filenames(alt_path)
+  errors = []
+
+  for file in filenames:
+            
+    content = open(file).read()
+    try:
+      metadata = parse_technique_metadata(content)
+      all_metadata[metadata['bundle_name']] = metadata
+
+      if include_methods_calls:
+        method_calls = parse_technique_methods(file)
+        all_metadata[metadata['bundle_name']]['method_calls'] = method_calls
+    except NcfError as e:
+      errors.append(e)
```

Comment:
That would be awesome if the filename can be added:

      error = NcfError("Could not parse Technique file " + file+", cause is: "+ e.message, e.details)
      errors.append(error)